### ISSUE TITLE
Error while running preprocess.log_transform()

### ISSUE THREAD

**Issue by kxxxjo** [dpeerlab/Palantir#75]
Hi, @ManuSetty 

Thanks for developing this useful tools!

When I run preprocess.log_transform(adata), there was the following error:

```
NotImplementedError                       Traceback (most recent call last)
<ipython-input-182-4af06e493e52> in <module>
----> 1 palantir.preprocess.log_transform(adata, pseudo_count=0.1)

~/opt/miniconda3/envs/scanpy/lib/python3.8/site-packages/palantir/preprocess.py in log_transform(data, pseudo_count)
     41     """
     42     if type(data) is sc.AnnData:
---> 43         data.X = np.log2(data.X + pseudo_count) - np.log2(pseudo_count)
     44     else:
     45         return np.log2(data + pseudo_count)

~/opt/miniconda3/envs/scanpy/lib/python3.8/site-packages/scipy/sparse/base.py in __add__(self, other)
    408                 return self.copy()
    409             # Now we would add this scalar to every element.
--> 410             raise NotImplementedError('adding a nonzero scalar to a '
    411                                       'sparse matrix is not supported')
    412         elif isspmatrix(other):

NotImplementedError: adding a nonzero scalar to a sparse matrix is not supported
```

To fix this error, I try these command:

```
scipy.sparse.csr_matrix(adata.X)
```

but the error still occurred.

could you help me to fix this error please?

and I have another question for reproducibility. I set random.seed(10) for reproducibility but the result of palantir was different every time. so what should I do to maintain the results?

Thanks.

Best,
Kyungjo

**Comment by rahulnutron** [comment]
Hi Kyungjo,

Have you got any solution, I am getting the same error.

Rahul

**Comment by kxxxjo** [comment]
Hi Rahul,

No I am still struggling to solve this problem.

Best,
KJ

**Comment by rahulnutron** [comment]
Hi KJ, Try this. Although not sure if this is a correct solution
ad.X = scipy.sparse.csr_matrix.todense(ad.X)

**Comment by ManuSetty** [comment]
Sorry for the late response. I have to put in a fix 19f30687b82da46e2e3e8d2d527b55ffe12fcaf5. The suggestion @rahulnutron  will work but will create a dense matrix increasing the memory requirement. Please pull the latest version of the repo for the fix.

**Comment by kxxxjo** [comment]
I used the latest version so the problem was solved! 

thanks for solution.

Thanks!

Best,
KJ
